### ISSUE TITLE
Support PEP 517 builds

### ISSUE THREAD

**Issue by Zaczero** [uber/h3-py#340]
When attempting to install the h3:

Note: This error originates from the build backend, and is likely not a problem with poetry but with h3 (3.7.6) not supporting PEP 517 builds. You can verify this by running 'pip wheel --no-cache-dir --use-pep517 "h3 (==3.7.6)"'.

**Comment by ajfriend** [comment]
Yes. I'd like to improve the current build setup, possibly using https://scikit-build-core.readthedocs.io/

PRs more than welcome in the meantime!

**Comment by wingkitlee0** [comment]
I was reading "modern build setup" these days.
It seems that the Cython support in `scikit-build-core` is still in development: https://scikit-build-core.readthedocs.io/en/latest/getting_started.html#source-code but it may have everything we need?

I was looking into `meson`/`meson-python` as well. There are currently used in `scipy` and `scikit-image`.
https://meson-python.readthedocs.io/en/latest/

**Comment by ajfriend** [comment]
Do you know if `meson` works well with CMake? The H3 C library uses CMake, so we need to interface with it. IIRC, this is why we were using `scikit-build` originally.

Aside from `meson`, `scikit-build-core` does look promising, I just haven't had a chance to try building with it.
